### DISCUSSION TITLE
Maximum number of device resources

### DISCUSSION THREAD

**Discussion by Rakshithakrishna2** [edgexfoundry/community#7] [General]
Hi all,

For my project I wanted to included sensors of different protocols in Device-service-c (EdgeX Jakarta).
For the same Can I please know how many max number of device resources can we include..?

Please help me in knowing.

Thank you,

**Comment by cloudxxx8** [comment]
There is no sepcific limitation for the number of device resources.  We verified around 2 thousands device resources, and it still works fine.

**Reply by iain-anderson** [reply]
Note that in the C implementation the resources are held as a simple list, so at very high cardinalities performance may begin to suffer.
